reactor-extra/src/main/java/reactor/retry/RetryBuilder.java

```diff
@@ -0,0 +1,440 @@
+	 *
+	 * @param backoffInterval delay between retries
+	 * @return updated {@link RetryBuilder}
+	 */
+	public RetryBuilder fixedBackoff(Duration backoffInterval) {
+		this.backoffCalculator = iteration -> iteration < maxAttempts ? backoffInterval : NO_RETRY;
+		return this;
+	}
+
+	/**
+	 * Configures an exponential backoff interval between retries. The scheduler used for backoff may
+	 * be configured usimng {@link #backoffScheduler(Scheduler)}. Retries are performed after a backoff
+	 * interval of <code>firstBackoff * (2 ** n)</code> where n is the number of retries completed (attempts -1).
+	 * If <code>maxBackoff</code> is not null, the maximum backoff applied will be limited to <code>maxBackoff</code>.
+	 * The total number of attempts can be limited by {@link #timeout(Duration)} and/or {@link #maxAttempts(int)}.
+	 *
+	 * @param firstBackoff delay for the first backoff, which is also used as the coefficient for subsequent backoffs
+	 * @param maxBackoff the maximum backoff delay before a retry
+	 * @return updated {@link RetryBuilder}
+	 */
+	public RetryBuilder exponentialBackoff(Duration firstBackoff, Duration maxBackoff) {
+		if (firstBackoff == null || firstBackoff.isNegative() || firstBackoff.isZero())
+			throw new IllegalArgumentException("firstBackoff must be > 0");
+		Duration maxBackoffInterval = maxBackoff != null ? maxBackoff : Duration.ofSeconds(Long.MAX_VALUE);
+		this.backoffCalculator = iteration -> {
+			Duration expBackoff = firstBackoff.multipliedBy((long) Math.pow(2, (iteration - 1)));
+			Duration backoff = expBackoff.compareTo(maxBackoffInterval) < 0 ? expBackoff : maxBackoffInterval;
+			return backoff;
+		};
+		return this;
+	}
+
+	/**
+	 * Configures a random de-correlated jitter backoff interval between retries. The scheduler used for backoff may
+	 * be configured using {@link #backoffScheduler(Scheduler)}. Retries are performed after a backoff
+	 * interval of <code>random_between(firstBackoff, prevBackoff * 3)</code>, with a minimum value of <code>firstBackoff</code>.
+	 * If <code>maxBackoff</code> is not null, the maximum backoff applied will be limited to <code>maxBackoff</code>.
+	 * The total number of attempts can be limited by {@link #timeout(Duration)} and/or {@link #maxAttempts(int)}.
+	 *
+	 * @param firstBackoff delay for the first backoff, also used as minimum backoff
+	 * @param maxBackoff the maximum backoff delay before a retry
+	 * @return updated {@link RetryBuilder}
+	 */
+	public RetryBuilder randomBackoff(Duration firstBackoff, Duration maxBackoff) {
+		if (firstBackoff == null || firstBackoff.isNegative() || firstBackoff.isZero())
+			throw new IllegalArgumentException("firstBackoff must be > 0");
+		long firstBackoffMs = firstBackoff.toMillis();
+		long maxBackoffMs = maxBackoff != null ? maxBackoff.toMillis() : Long.MAX_VALUE;
+		if (maxBackoffMs < firstBackoffMs)
+			throw new IllegalArgumentException("maxBackoff must be > firstBackoff");
+		AtomicLong prevBackoffMs = new AtomicLong();
```

Comment:
I'm also wondering, since we have some state, if it shouldn't be encapsulated into some `Jitter` instance right away. Or if we'd need a concrete class with the state rather than a lambda...